File: Leetcode/CPP/moutainpeek.cpp
#include <iostream>
#include <math.h>
#include <vector>
using namespace std;

//find the peek index of the moutain array in cpp
int peek(vector<int> arr){
    int start = 0; 
    int end = arr.size() - 1;
    int mid = start + (end - start) / 2;
    while(start < end){

        if (arr[mid] < arr[mid + 1]){
            start = mid + 1;
        }

        else{
            end = mid;
        }

        mid = start + (end - start) / 2;
    }

    return mid;
}

int main(){

    vector <int> arrr = {3,4,5,1};

    int value = peek(arrr);

    cout << value << endl;

}

File: Leetcode/Python/moutainpeek.py
'''
find the peek index of the mountain array in python.
'''

def peek(arr, size):

    start = 0
    end = size - 1

    while (start < end):
        mid = start + (end - start) // 2

        if arr[mid] < arr[mid + 1]:
            start = mid + 1
        else:
            end = mid

    return mid


if __name__ == '__main__':
    arr = [3,4,5,1]
    size = len(arr)

    print(peek(arr, size))
